packages/braintree-integration/src/braintree-fastlane/braintree-fastlane-customer-strategy.ts

```diff
@@ -0,0 +1,145 @@
+import { BraintreeInitializationData } from '@bigcommerce/checkout-sdk/braintree-utils';
+import {
+    CustomerCredentials,
+    CustomerInitializeOptions,
+    CustomerStrategy,
+    ExecutePaymentMethodCheckoutOptions,
+    InvalidArgumentError,
+    PaymentIntegrationService,
+    PaymentMethod,
+    RequestOptions,
+} from '@bigcommerce/checkout-sdk/payment-integration-api';
+
+import { WithBraintreeFastlaneCustomerInitializeOptions } from './braintree-fastlane-customer-initialize-options';
+import BraintreeFastlaneUtils from './braintree-fastlane-utils';
+
+export default class BraintreeFastlaneCustomerStrategy implements CustomerStrategy {
+    private isAcceleratedCheckoutEnabled = false;
+    private isFastlaneEnabled = false;
+
+    constructor(
+        private paymentIntegrationService: PaymentIntegrationService,
+        private braintreeFastlaneUtils: BraintreeFastlaneUtils,
+    ) {}
+
+    async initialize({
+        methodId,
+        braintreeacceleratedcheckout,
+    }: CustomerInitializeOptions &
+        WithBraintreeFastlaneCustomerInitializeOptions): Promise<void> {
+        if (!methodId) {
+            throw new InvalidArgumentError(
+                'Unable to proceed because "methodId" argument is not provided.',
+            );
+        }
+
+        const paymentMethod = await this.getValidPaymentMethodOrThrow(methodId);
+
+        this.isAcceleratedCheckoutEnabled =
+            !!paymentMethod.initializationData?.isAcceleratedCheckoutEnabled;
+
+        this.isFastlaneEnabled =
+            !!paymentMethod.initializationData?.isFastlaneEnabled;
+
+        if (!this.isFastlaneEnabled) {
+            if (this.isAcceleratedCheckoutEnabled) {
+                await this.braintreeFastlaneUtils.initializeBraintreeConnectOrThrow(
+                    paymentMethod.id,
+                    braintreeacceleratedcheckout?.styles,
+                );
+            }
+        } else {
+            await this.braintreeFastlaneUtils.initializeBraintreeFastlaneOrThrow(
+                paymentMethod.id,
+                braintreeacceleratedcheckout?.styles,
+            );
+        }
```

Comment:
First of all we should check if Accelerated Checkout feature is enabled. If its not, then we should not initialise Braintree Connect or Fastlane at all. Then if Fastlane is enabled we should call initializeBraintreeFastlaneOrThrow otherwise  initializeBraintreeConnectOrThrow